### PR TITLE
Add all pebble sstable tests, support 2-level index SSTables

### PR DESCRIPTION
**Pull request by arielshaqed**

### CODE REVIEW

**Comment by arielshaqed** [comment]
This *adds* to #2335 -- reviewer's choice in which order to review.

**Comment by arielshaqed** [comment]
@itaiad200 @talSofer please review this one too.  It does not make sense to pull #2335 without this one.  Reviewing this one is not "required", but that's just because of a weirdness in GitHub branch protection.

File: clients/spark/core/src/main/scala/io/treeverse/jpebble/BlockParser.scala [hunk 1]
```diff
@@ -250,54 +258,76 @@ object BlockParser {
...
+   */
+  def entryIterator(in: BlockReadable): Iterator[Entry] = {
+    val bytes = in.iterate(in.length - BlockParser.footerLength, BlockParser.footerLength)
+    val footer = BlockParser.readFooter(bytes)
+    BlockParser.readEnd(bytes)
+
+    val blockIndexType = {
+      val props = BlockParser.readProperties(in, footer)
+      val typeIt = props(BlockParser.INDEX_TYPE_KEY).iterator
+      val typ = BlockParser.readFixedInt(typeIt)
```
Comment (itaiad200) [inline clients/spark/core/src/main/scala/io/treeverse/jpebble/BlockParser.scala:294]: Why an iterator? Can there be more than a single value for a key in the props?

**Review by itaiad200 — APPROVED** [review]

File: clients/spark/core/src/main/scala/io/treeverse/jpebble/BlockParser.scala [hunk 2]
```diff
@@ -250,54 +258,76 @@ object BlockParser {
...
+      BlockParser.parseDataBlock(block)
+    })
 
-  private def stripInternalKey(entry: Entry) =
-    new Entry(entry.key.slice(0, entry.key.length - 8), entry.value)
-
-  def hasNext = advanceToEntry()
-
-  def next(): Entry = stripInternalKey(entryIt.next)
+    entryIt.map({ case Entry(key, value) => new Entry(key.slice(0, key.length - 8), value) })
```
[the remark this answers is not in the document]
Comment (arielshaqed) [reply clients/spark/core/src/main/scala/io/treeverse/jpebble/BlockParser.scala]: Already commented & fix pushed on #2335. :-)

File: clients/spark/core/src/test/scala/io/treeverse/jpebble/BlockParserSpec.scala [hunk 3]
```diff
@@ -156,68 +157,90 @@ class BlockParserSpec extends AnyFunSpec with Matchers {
...
+      "h.table-bloom.no-compression.sst",
+      "h.table-bloom.sst"
+    )
 
-      val bh = footer.metaIndex
-      Console.out.println("[metaindex] " +
-        f"${bh.offset}%08x (${bh.offset}) -> ${bh.size}%08x (${bh.size}%d)")
+    /**
+     * Lightweight fixture for running particular tests with an SSTable and
+     * recovering its handle aafter running.  See
```
[the remark this answers is not in the document]
Comment (arielshaqed) [reply clients/spark/core/src/test/scala/io/treeverse/jpebble/BlockParserSpec.scala]: Fine, I'll use just one 'a' with no redundancy.

**Review by arielshaqed — COMMENTED** [review]
Thanks!

**Comment by arielshaqed** [comment]
Pulling (once tested) as approved.  THANKS!

**Pull request merged**